includes/class-wc-subscriptions-order.php

```diff
@@ -622,36 +622,40 @@ public static function get_users_subscription_orders( $user_id = 0 ) {
 		}
 
 		// Get all the customers orders which are not subscription renewal orders
-		$order_ids = get_posts( array(
-			'posts_per_page' => 1,
-			'post_type'      => 'shop_order',
-			'post_status'    => 'any',
-			'fields'         => 'ids',
-			'orderby'        => 'date',
-			'order'          => 'DESC',
-			'meta_query'     => array(
-				array(
-					'key'     => '_customer_user',
-					'compare' => '=',
-					'value'   => $user_id,
-					'type'    => 'numeric',
-				),
-				array(
+		$custom_query_var_handler = function( $query, $query_vars ) {
+			if ( ! empty( $query_vars['_non_subscription_renewal'] ) ) {
+				$query['meta_query'][] = array(
 					'key'     => '_subscription_renewal',
 					'compare' => 'NOT EXISTS',
-				),
-			),
-		) );
+				);
+			}
 
-		foreach ( $order_ids as $index => $order_id ) {
-			if ( ! wcs_order_contains_subscription( $order_id, 'parent' ) ) {
-				unset( $order_ids[ $index ] );
+			return $query;
+		};
+		add_filter( 'woocommerce_order_data_store_cpt_get_orders_query', $custom_query_var_handler, 10, 2 );
+
+		$args = array(
+			'limit'       => 1,
+			'type'        => 'shop_order',
+			'status'      => 'any',
+			'orderby'     => 'date',
+			'order'       => 'DESC',
+			'customer_id' => $user_id,
+		);
+
+		$args['_non_subscription_renewal'] = true;
```

Comment:
I use a unique string `_non_subscription_renewal` here, so I can check for it in `$custom_query_var_handler`.

Reply:
Makes sense!

I wonder if we're going to see a number of duplicated meta query filters being used for the same filter. If we do start seeing this, I curious to know what we could do to prevent duplicated code.

Something like a new file `wcs-order-meta-query-filters.php` which has a bunch of public functions like `wcs_filter_orders_non_subscription_renewal()` so we can call them from anywhere.

I don't think we should implement this just yet but just thinking out aloud while reviewing the changes here :)

Nice one!

Reply:
That's a good point.

I can't think of any other way. Seems like we will do a lot of:
```
add_filter( 'woocommerce_order_data_store_cpt_get_orders_query', $callable );
wc_get_orders( $args );
remove_filter( 'woocommerce_order_data_store_cpt_get_orders_query', $callable );
```

We can have a 'central' place for the callables and just add filter once, per your suggestion with `wcs-order-meta-query-filters.php` (if I understand it correctly) but the downside is that part of the code will not be 'close' to the relevant code that calls `wc_get_orders()`.

Reply:
If this is a repeated pattern we should look at how to remove the repetition. This is likely outside the scope of the current work but would aid in readability and would simplify code.
```php
function filter_get_orders( $args , $callable ) {
    add_filter( 'woocommerce_order_data_store_cpt_get_orders_query', $callable ); 
    $orders = wc_get_orders( $args ); 
    remove_filter( 'woocommerce_order_data_store_cpt_get_orders_query', $callable );
    return $orders;
}

$orders = filter_get_orders(
    $args,
    function ( $query ) {
       // alter query here
    }
);
```

Reply:
@brucealdridge after giving it a thought, though your suggestion is a good idea, I'm afraid this repetition won't come up again and in that case, having a helper function as suggested would be over-engineering. We can address it later as needed.